plugins/woocommerce/includes/wc-core-functions.php

```diff
@@ -2033,9 +2033,10 @@ function wc_add_number_precision( ?float $value, bool $round = true ) {
 		return 0.0;
 	}
 
-	$cent_precision = pow( 10, wc_get_price_decimals() );
-	$value          = $value * $cent_precision;
-	return $round ? NumberUtil::round( $value, wc_get_rounding_precision() - wc_get_price_decimals() ) : $value;
+	$result          = $value * pow( 10, wc_get_price_decimals() );
+	$round_precision = $round ? wc_get_rounding_precision() - wc_get_price_decimals() : wc_get_rounding_precision();
```

Comment:
https://github.com/woocommerce/woocommerce/issues/26654 re-appeared on PHP 8.4

Reply:
@jorgeatorres @Konamiman : I'd love to hear your thoughts on this change (fixes a regression on PHP 8.4) - both on technical side and which core version it should target.